tests/test_paths_counts.cpp

```diff
@@ -0,0 +1,63 @@
+#include "gtest/gtest.h"
+
+#include <graph/paths_counts.h>
+#include <graph/adjacency_matrix.h>
+#include <graph/flatten_adjacency_matrix.h>
+
+#include <vector>
+
+template <typename T>
+class PathCountsTest : public testing::Test {
+};
+
+using GraphImplementations = ::testing::Types<
+  AdjacencyMatrix<std::uint8_t>,
+  FlattenAdjacencyMatrix<bool>
+>;
+
+TYPED_TEST_SUITE(PathCountsTest, GraphImplementations,);
+
+using Matrix2d = std::vector<std::vector<std::uint32_t>>;
+
+// Helper function to test equality of all edges
+template<typename GraphT>
+void ASSERT_GRAPH_EQ(const GraphT& graph, size_t n_nodes, const Matrix2d& target_matrix){
```

Comment:
I think it is better to implement `operator==` for the `class Graph`

Reply:
Done, with test